src/test/ensemble_flow_custom_node_tests.cpp

```diff
@@ -980,4 +997,461 @@ TEST_F(EnsembleFlowCustomNodeLoadConfigThenExecuteTest, ReferenceLibraryWithRest
+            switch (j) {
+            case ADD:
+                expectedOutput[index] = input[i] + factors[j];
+                break;
+            case SUB:
+                expectedOutput[index] = input[i] - factors[j];
+                break;
+            case MULTIPLY:
+                expectedOutput[index] = input[i] * factors[j];
+                break;
+            case DIVIDE:
+                expectedOutput[index] = input[i] / factors[j];
+                break;
+            }
+        }
+    }
+}
+
+enum class Method {
+    MAXIMUM_MAXIMUM,
+    MAXIMUM_MINIMUM,
+    MAXIMUM_AVERAGE,
+};
+
+std::vector<float> prepareGatherHighestExpectedOutput(std::vector<float> input, Method option) {
+    const size_t dummy_size = 10;
+    std::vector<float> expectedOutput(dummy_size);
+    size_t tensorsCount = input.size() / dummy_size;
+    // perform operations
+    std::vector<float> minimums(tensorsCount, std::numeric_limits<int>::max());
+    std::vector<float> maximums(tensorsCount, std::numeric_limits<int>::lowest());
+    std::vector<float> averages(tensorsCount, 0);
+    for (size_t opId = 0; opId < tensorsCount; ++opId) {  // iterate over ops
+        for (size_t i = 0; i < dummy_size; ++i) {
+            size_t index = dummy_size * opId + i;
+            switch (option) {
+            case Method::MAXIMUM_MAXIMUM:
+                maximums[opId] = std::max(maximums[opId], input[index]);
+                break;
+            case Method::MAXIMUM_MINIMUM:
+                minimums[opId] = std::min(maximums[opId], input[index]);
+                break;
+            case Method::MAXIMUM_AVERAGE:
+                averages[opId] += input[index];
+                break;
+            default:
+                throw std::logic_error("");
+                break;
+            }
+        }
+        averages[opId] /= dummy_size;
+    }
+    // choose tensor
+    size_t whichTensor = 42;
+    const std::vector<float>* fromWhichContainerToChoose = &maximums;
+    switch (option) {
+    case Method::MAXIMUM_MAXIMUM:
+        fromWhichContainerToChoose = &maximums;
+        break;
+    case Method::MAXIMUM_MINIMUM:
+        fromWhichContainerToChoose = &minimums;
+        break;
+    case Method::MAXIMUM_AVERAGE:
+        fromWhichContainerToChoose = &averages;
+        break;
+    default:
+        throw std::logic_error("");
+    }
+    whichTensor = std::distance(fromWhichContainerToChoose->begin(),
+        std::max_element(fromWhichContainerToChoose->begin(),
+            fromWhichContainerToChoose->end()));
+    // copy tensor
+    std::copy(input.begin() + dummy_size * whichTensor,
+        input.begin() + dummy_size * (whichTensor + 1),
+        expectedOutput.begin());
+    return expectedOutput;
+}
+
+TEST_F(EnsembleFlowCustomNodeAndDemultiplexerLoadConfigThenExecuteTest, JustDifferentOpsCustomNode) {
+    std::unique_ptr<Pipeline> pipeline;
+    std::cout << pipelineCustomNodeDifferentOperationsConfig << std::endl;
```

Comment:
Doesnt ->loadConfiguration create file with content therefore print config anyway?

Reply:
Yes - leftover debug. To be removed.